src/WebJobs.Extensions.ApiHub/Common/GenericFileBinding.cs

```diff
@@ -0,0 +1,145 @@
+            Func<Task> onComplete;
+
+            if ((_source.Access == FileAccess.Write) || (targetType == typeof(TextWriter)))
+            {
+                var tuple = await strategy.OpenWriteStreamAsync();
+                var writeStream = tuple.Item1;
+                var saveStreamFunc = tuple.Item2;
+
+                if (targetType == typeof(Stream))
+                {
+                    userObj = writeStream;
+                    onComplete = saveStreamFunc;
+                }
+                else if (targetType == typeof(TextWriter))
+                {
+                    var tw = new StreamWriter(writeStream);
+                    userObj = tw;
+                    onComplete = async () =>
+                    {
+                        tw.Flush();
+                        await saveStreamFunc();
+                    };
+                }
+                else
+                {
+                    // $$$ Move this error up to the binder level.
+                    // Unknown?
+                    throw new InvalidOperationException("Unspported type:" + targetType.FullName);
+                }
+            }
+            else
+            {
+                var readStream = await strategy.OpenReadStreamAsync();
+                onComplete = () => Task.FromResult(0); // Nop
+                if (targetType == typeof(Stream))
+                {
+                    userObj = targetType;
+                }
+                else if (targetType == typeof(TextReader))
+                {
+                    userObj = new StreamReader(readStream);
+                }
+                else if (targetType == typeof(string))
+                {
+                    userObj = new StreamReader(readStream).ReadToEnd();
+                }
+                else 
+                {
+                    throw new InvalidOperationException("Unspported type:" + targetType.FullName);
+                }
+            }
+                        
+            
+            //var text = await Convert(targetType, _strategy, path);
+
+            var valueProvider = new ConstantObj
+            {
+                Type = targetType,
+                _value = userObj,
+                _onCompleted = onComplete
+            };
+            return valueProvider;
+        }
+
+
+        // Get a stream binder from a resolved path
+        private async Task<IFileStreamProvider> GetStreamHelper(string path)
+        {
+            TAttribute clone = JsonConvert.DeserializeObject<TAttribute>(JsonConvert.SerializeObject(_source));
+            clone.Path = path;
+
+            TFile nativeFile = await this._strategyBuilder(clone);
+            var func = this._converterManager.GetConverter<TFile, IFileStreamProvider>();
+            IFileStreamProvider strategy = func(nativeFile);
+            return strategy;
+        }
+
+        public ParameterDescriptor ToParameterDescriptor()
+        {
+            return new ParameterDescriptor { };
```

Comment:
Add a TODO here to finish Dashboard integration for your binding. At a minimum, this needs to set the parameter Name - the dashboard will likely error without it.